src/vw/src/main/scala/VowpalWabbitFeaturizer.scala

```diff
@@ -0,0 +1,140 @@
+
+  val stringSplitInputCols = new StringArrayParam(this, "stringSplitInputCols",
+    "Input cols that should be split add word boundaries")
+  setDefault(stringSplitInputCols -> Array())
+
+  def getStringSplitInputCols: Array[String] = $(stringSplitInputCols)
+  def setStringSplitInputCols(value: Array[String]): this.type = set(stringSplitInputCols, value)
+
+  private def getAllInputCols = getInputCols ++ getStringSplitInputCols
+
+  private def getFeaturizer(name: String, dataType: DataType, idx: Int, namespaceHash: Int): Featurizer = {
+    val stringSplitInputCols = getStringSplitInputCols
+
+    dataType match {
+      case DoubleType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getDouble(idx))
+      case FloatType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getFloat(idx).toDouble)
+      case IntegerType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getInt(idx).toDouble)
+      case LongType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getLong(idx).toDouble)
+      case ShortType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getShort(idx).toDouble)
+      case ByteType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getByte(idx).toDouble)
+      case BooleanType => new BooleanFeaturizer(idx, name, namespaceHash, getMask)
+      case StringType =>
+        if (stringSplitInputCols.contains(name))
+          new StringSplitFeaturizer(idx, name, namespaceHash, getMask)
+      else new StringFeaturizer(idx, name, namespaceHash, getMask)
+      case arr: ArrayType => {
+        if (arr.elementType != DataTypes.StringType)
+          throw new RuntimeException(s"Unsupported array element type: ${dataType}")
+
+        new StringArrayFeaturizer(idx, name, namespaceHash, getMask)
+      }
+      case m: MapType => {
+        if (m.keyType != DataTypes.StringType)
+          throw new RuntimeException(s"Unsupported map key type: ${dataType}")
+
+        m.valueType match {
+          case StringType => new MapStringFeaturizer(idx, name, namespaceHash, getMask)
+          case DoubleType => new MapFeaturizer[Double](idx, name, namespaceHash, getMask, v => v)
+          case FloatType => new MapFeaturizer[Float](idx, name, namespaceHash, getMask, v => v.toDouble)
+          case IntegerType => new MapFeaturizer[Int](idx, name, namespaceHash, getMask, v => v.toDouble)
+          case LongType => new MapFeaturizer[Long](idx, name, namespaceHash, getMask, v => v.toDouble)
+          case ShortType => new MapFeaturizer[Short](idx, name, namespaceHash, getMask, v => v.toDouble)
+          case ByteType => new MapFeaturizer[Byte](idx, name, namespaceHash, getMask, v => v.toDouble)
+          case _ => throw new RuntimeException(s"Unsupported map value type: ${dataType}")
+        }
+      }
+      case m: Any =>
+        if (m.typeName == "vector") // unfortunately the type is private
```

Comment:
We have a namespace injection for this

Reply:
you have a pointer to an example for me?

Reply:
https://github.com/Azure/mmlspark/blob/master/src/main/scala/com/microsoft/ml/spark/stages/EnsembleByKey.scala

See VectorType